lib/cc/engine/analyzers/violation.rb

```diff
@@ -0,0 +1,90 @@
+module CC
+  module Engine
+    module Analyzers
+      class Violation
+        BASE_POINTS = 10_000
+
+        attr_reader :issue
+
+        def initialize(issue, hashes, directory)
+          @issue = issue
+          @hashes = hashes
+          @directory = directory
+        end
+
+        def format
+          {
+            "type": "issue",
+            "check_name": name,
+            "description": "Duplication found in #{issue.name}",
+            "categories": ["Duplication"],
+            "location": format_location,
+            "remediation_points": calculate_points,
+            "other_locations": format_other_locations,
+            "content": content_body
+          }
+        end
+
+        private
+
+        attr_reader :hashes, :directory
+
+        def current_sexp
+          @location ||= hashes.first
+        end
+
+        def other_sexps
+          @other_locations ||= hashes.drop(1)
```

Comment:
It's from `Flay#hashes` which gives you a hash of `hash_string -> [s-expression]`. We drop here because these are `other_locations` to report.